nx-X11/programs/Xserver/dix/window.c

```diff
@@ -1846,7 +1845,23 @@ ResizeChildrenWinSize(register WindowPtr pWin, int dx, int dy, int dw, int dh)
 	pSib->drawable.y = pWin->drawable.y + pSib->origin.y;
 	SetWinSize (pSib);
 	SetBorderSize (pSib);
+
+#ifdef NXAGENT_SERVER
+        /*
+         * Don't force X to move children. It will position them
+         * according with gravity.
+         *
+         * (*pScreen->PositionWindow)(pSib, pSib->drawable.x, pSib->drawable.y);
+         */
+
+        /*
+         * Update pSib privates, as this window is moved by X.
+         */
+
+        nxagentAddConfiguredWindow(pSib, CW_Update);
+#else
 	(*pScreen->PositionWindow)(pSib, pSib->drawable.x, pSib->drawable.y);
+#endif
```

Comment:
Copyright problem here. Don't mix GPL-2 code with MIT/X11 (or some such) code. The above code fragment needs to stay in hw/nxagent/.